local rate_limit support match multiple descriptors (#20869)
envoyproxy/envoy: docs/root/configuration/http/http_filters/local_rate_limit_filter.rst

settings are then used to decide if the request should be rate limited or not
depending on whether the local descriptor's entries match the route's rate
limit actions descriptor entries. If there is no matching descriptor entries,
the default token bucket is used. All the matched local descriptors will try
to consume tokens, and in most cases if one of them is limited, the remaining

> descriptors will not consume their tokens.

Reviewer: Can you more clearly document the sorting and the possible correctness implications?

Also this change will need a release note.

settings are then used to decide if the request should be rate limited or not
depending on whether the local descriptor's entries match the route's rate
limit actions descriptor entries. If there is no matching descriptor entries,
the default token bucket is used. All the matched local descriptors will be
sorterd by tokens per second and try to consume tokens in order, in most cases
if one of them is limited, the remaining descriptors will not consume their tokens.
However, in some cases, it may not work, for example, we have two descriptors
A and B, A is limited 3 requests per second, and B is limited 20 requests per 10 seconds.
Obviously B is stricter than A (token per second), as a result, if we send requests

> above 3 in a second, the limited requests form A will also consume tokens of B.

Reviewer: ```suggestion
above 3 in a second, the limited requests from A will also consume tokens of B.
```

> Since global token is not sorted, so we suggest it should be larger than other descriptors.

Reviewer: ```suggestion
Note that global tokens are not sorted, so we suggest they should be larger than other descriptors.
```

Author: done, thanks, PTAL again @mattklein123
